### ISSUE TITLE
🐛 useLiteralKeys breaks object key

### ISSUE THREAD

**Issue by seriousme** [biomejs/biome#191]
### Environment information

```block
CLI:
  Version:                      1.1.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.17.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true
```

### What happened?

```javascript
const obj = { "": "blah"}
```
results in: 
`The computed expression can be simplified without the use of a string literal.`

Apply fixes results in:

```javascript
const obj = {  : "blah"}
```

Which is invalid code.

https://biomejs.dev/playground/?code=YwBvAG4AcwB0ACAAbwBiAGoAIAA9ACAAewAgACIAIgA6ACAAIgBiAGwAYQBoACIAfQA%3D

### Expected result

The rule should not trigger on object keys.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ssssota** [comment]
Can I try this issue?

**Comment by Conaclos** [comment]
@ssssota You are welcome! Assigned :)

**Comment by seriousme** [comment]
Thx !
